File: SortingAlgorithm/_24_OddEvenSort/OddEvenSort.java
package Algorithm.SortingAlgorithm._24_OddEvenSort;

public class OddEvenSort {
    public static void oddEvenSort(int arr[]) {
        int n = arr.length;
        boolean isSorted = false; // Initially, the array is unsorted

        while (!isSorted) {
            isSorted = true;
            int temp = 0;

            // Perform Bubble sort on odd-indexed elements
            for (int i = 1; i <= n - 2; i += 2) {
                if (arr[i] > arr[i + 1]) {
                    temp = arr[i];
                    arr[i] = arr[i + 1];
                    arr[i + 1] = temp;
                    isSorted = false;
                }
            }

            // Perform Bubble sort on even-indexed elements
            for (int i = 0; i <= n - 2; i += 2) {
                if (arr[i] > arr[i + 1]) {
                    temp = arr[i];
                    arr[i] = arr[i + 1];
                    arr[i + 1] = temp;
                    isSorted = false;
                }
            }
        }
    }

    public static void main(String[] args) {
        int arr[] = {34, 2, 10, -9};
        int n = arr.length;

        oddEvenSort(arr);

        System.out.print("Sorted array: ");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
}

File: SortingAlgorithm/_24_OddEvenSort/OddEvenSort.py
def odd_even_sort(arr):
    """
    Sorts an array using the Odd-Even (or Brick) sort algorithm.

    :param arr: List of elements to be sorted.
    """
    n = len(arr)
    is_sorted = False

    while not is_sorted:
        is_sorted = True
        for i in range(1, n - 1, 2):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                is_sorted = False

        for i in range(0, n - 1, 2):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                is_sorted = False


if __name__ == "__main__":
    arr = [34, 2, 10, -9]
    odd_even_sort(arr)

    print("Sorted array:")
    for num in arr:
        print(num, end=" ")
